### ISSUE TITLE
Start Playback before Song is fully Downloaded

### ISSUE THREAD

**Issue by funkyg** [funkyg/funkytunes#2]

**Comment by heyakyra** [comment]
would code from https://github.com/mpv-android/mpv-android be useful? 

or https://github.com/popcorn-official/popcorn-android

or https://code.videolan.org/videolan/vlc-android

**Comment by funkyg** [comment]
yes those seem useful

https://github.com/TorrentStream/TorrentStream-Android seems to be the library used by popcorntime android, so its probably the best option

**Comment by heyakyra** [comment]
also may be worth looking into torrenttunes: 

https://github.com/dessalines/torrenttunes-android
https://github.com/dessalines/torrenttunes-client

**Comment by funkyg** [comment]
just looked at the code for torrenttunes-android, doesnt look like it can play anything.

**Comment by funkyg** [comment]
i just noticed that jlibtorrent also has a sequential download mode. so we could just use that and start playback when the file reaches a certain percentage

http://www.libtorrent.org/reference-Core.html#set_sequential_download()

**Comment by gjedeer** [comment]
> In this mode, piece priorities are ignored, with the exception of priority 7, which are still preferred over the sequential piece order.

Doesn't it mean it will also download files sequentially (track1, track2, track3, ...) ignoring what the user chose?

**Comment by funkyg** [comment]
no, pieces are parts of an individual file. so its independent from the file order

https://en.wikipedia.org/wiki/Torrent_file#Background
